src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitXmlIterator.java

```diff
@@ -261,24 +261,34 @@ private void handleJUnitTest(XMLEvent event) throws XMLStreamException, IOExcept
                     }
 
                     String cleanedTestName = cleanTestName(testName);
+                    String nodeName = "";
+                    if(!nodeNames.isEmpty()){
+                        nodeName = nodeNames.stream().findFirst().get();
+                    }
                     boolean testReportCreated = true;
                     if (additionalContext != null && additionalContext instanceof List) {
                         //test folders are appear in the following format GUITest1[1], while [1] number of test. It possible that tests with the same name executed in the same job
                         //by adding [1] or [2] we can differentiate between different instances.
                         //We assume that test folders are sorted so in this section, once we found the test folder, we remove it from collection , in order to find the second instance in next iteration
                         List<String> createdTests = (List<String>) additionalContext;
                         String searchFor = cleanedTestName + "[";
-                        Optional<String> optional = createdTests.stream().filter(str -> str.startsWith(searchFor)).findFirst();
+                        Optional<String> optional = createdTests.stream().filter(str -> str.contains(searchFor)).findFirst();
                         if (optional.isPresent()) {
-                            cleanedTestName = optional.get();
-                            createdTests.remove(cleanedTestName);
+                            String nodeTestString = optional.get();
+                            String node = nodeTestString.split("/")[0];
```

Comment:
can you please add check before split that you have "/" exist to prevent unplanned errors

Reply:
Added the check.